pkgs/by-name/ob/objfw/package.nix

```diff
@@ -0,0 +1,47 @@
+{
+  autoconf,
+  autogen,
+  automake,
+  clang,
+  fetchFromGitHub,
+  lib,
+  stdenv,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "objfw";
+  version = "1.1.7";
+
+  src = fetchFromGitHub {
+    owner = "ObjFW";
+    repo = "ObjFW";
+    rev = "1.1.7-release";
+    hash = "sha256-0ylG/2ZSO3b8zdh6W9QJH5OJW9V344CCik1DduV5mhI=";
+  };
+
+  nativeBuildInputs = [
+    clang
+    automake
+    autogen
+    autoconf
+  ];
+
+  buildPhase = ''
+    ./autogen.sh
+    ./configure --prefix=$out --without-tls
+    make
+  '';
+
+  doCheck = true;
+  checkPhase = "make check";
+
+  installPhase = "make install";
```

Comment:
I'm not really sure about this section. Though it is not my area of expertise, so I don't want to give you a blocking review either.

For now:
1. Move `./autogen.sh` and `./configure` to `configurePhase`. Add `runHook preConfigure` and `runHook postConfigure` at the beginning and at the end of the phase respectively.
2. Remove buildPhase, `make` should be invoked by mkDerivation automatically.
3. You can try omitting installPhase as well, but I am not very sure.

Reply:
Thanks, I've done those. It works fine without the `installPhase`, so I've omitted that